packages/web/src/components/comments/CommentSectionDesktop.tsx

```diff
@@ -13,18 +17,66 @@ export const CommentSectionDesktop = () => {
     currentUserId,
     comments,
     commentSectionLoading,
+    isLoadingMorePages,
+    reset,
+    hasMorePages,
     handleLoadMoreRootComments
   } = useCurrentCommentSection()
 
   const commentPostAllowed = currentUserId !== null
+  const commentSectionRef = useRef<HTMLDivElement | null>(null)
+
+  // Need refs for these values because the scroll handler will not be able to access state changes
+  const isLoadingMorePagesRef = useRef(isLoadingMorePages)
+  const hasMorePagesRef = useRef(hasMorePages)
+  useEffect(() => {
+    // Keep the ref values up to date
+    isLoadingMorePagesRef.current = isLoadingMorePages
+    hasMorePagesRef.current = hasMorePages
+  }, [isLoadingMorePages, hasMorePages])
+
+  // Endless scroll pagination logic
+  useEffectOnce(() => {
+    const handleScroll = (e: Event) => {
+      if (
+        !isLoadingMorePagesRef.current &&
+        hasMorePagesRef.current &&
+        commentSectionRef.current
+      ) {
+        const scrollTopPosition = (e.target as HTMLElement)?.scrollTop ?? 0 // How far along the window has scrolled
+        const containerOffsetHeight = commentSectionRef.current.offsetHeight // How tall our container is
+        // Start loading more at 75% of the way scrolled to the bottom
+        const shouldLoadMore = scrollTopPosition >= containerOffsetHeight * 0.75
+        if (shouldLoadMore) {
+          handleLoadMoreRootComments()
+        }
+      }
+    }
+    const listenerFn = debounce(handleScroll, 50) // small debounce to avoid a race condition between scroll handler and state updates to isLoadingMorePages/hasMorePages
+    window.addEventListener('scroll', listenerFn, true) // Note: 3rd true argument is to make sure we capture the event at the window level (otherwise it gets captured elsewhere)
```

Comment:
We have an `InfiniteScroll` component that is used in Lineup, idk if you tried using that already

Reply:
I didnt actually, lemme look into it; might be simpler

Reply:
Nice, feels a little simpler 👌